qn-7.js

```diff
@@ -0,0 +1,119 @@
+// number one
+const products = [
+  { name: "Product 1", count: 10, price: 100 },
+  { name: "Product 2", count: 5, price: 50 },
+  { name: "Product 3", count: 20, price: 200 },
+  { name: "Product 5", count: 0, price: 50 },
+
+];
+
+function getTotalItemCount() {
+  let totalCount = 0;
+  products.forEach((product) => {
+    totalCount += product.count;
+  });
+  console.log(totalCount);
+}
+getTotalItemCount();
+
+//   number two
+function createNewItem(name, count, price) {
+  const newItem = { name, count, price };
+  products.forEach(item => {
+  if(item.name !== newItem.name){
+  products.push(newItem);
+
+  }else{
+    console.log("the product already exist.");
+  }
+});
+ 
+  console.log("you added "+newItem.name +" sucsesfully" );
+}
+
+createNewItem("Product 4", 15, 150);
+//   number three
+function getAvailableItems() {
+  console.log(products.filter((product) => product.count > 0)); 
```

Comment:
Could you please explain the `filter` method here?

Reply:
the filter is actually like searching, it searches all  indexes and gives us what we wanted to found